src/main/java/com/checkmarx/flow/service/JiraService.java

```diff
@@ -957,91 +954,121 @@ private String getBody(ScanResults.XIssue issue, ScanRequest request, String fil
                         }
+                        createIssue(request, newIssues, xIssue, currentIssue);
                     }
                 }
             } catch (RestClientException e) {
                 log.error("Error occurred while processing issue with key {}", xIssue.getKey(), e);
                 throw new JiraClientException();
             }
+            log.debug("Issue: {} successfully updated. Removing it from dynamic scan results map", xIssue.getValue());
+            nonPublishedScanResultsMap.remove(xIssue.getKey());
         }
 
         /*Check if an issue exists in Jira but not within results and close if not*/
+        closeIssueInCaseNotWithinResults(request, map, jiraMap, closedIssues);
+
+        return ImmutableMap.of(
+                "new", newIssues,
+                "updated", updatedIssues,
+                "closed", closedIssues
+        );
+    }
+
+    private void closeIssueInCaseNotWithinResults(ScanRequest request, Map<String, ScanResults.XIssue> map, Map<String, Issue> jiraMap, List<String> closedIssues) throws JiraClientException {
         for (Map.Entry<String, Issue> jiraIssue : jiraMap.entrySet()) {
             try {
-                if (!map.containsKey(jiraIssue.getKey())) {
-                    if (request.getBugTracker().getOpenStatus().contains(jiraIssue.getValue().getStatus().getName())) {
-                        /*Close the issue*/
-                        log.info("Closing issue with key {}", jiraIssue.getKey());
-                        this.transitionCloseIssue(jiraIssue.getValue().getKey(),
-                                request.getBugTracker().getCloseTransition(), request.getBugTracker(), false); //No false positives
-                        closedIssues.add(jiraIssue.getValue().getKey());
-                    }
+                if (!map.containsKey(jiraIssue.getKey()) && (request.getBugTracker().getOpenStatus().contains(jiraIssue.getValue().getStatus().getName()))) {
+                    /*Close the issue*/
+                    log.info("Closing issue with key {}", jiraIssue.getKey());
+                    this.transitionCloseIssue(jiraIssue.getValue().getKey(),
+                            request.getBugTracker().getCloseTransition(), request.getBugTracker(), false); //No false positives
+                    closedIssues.add(jiraIssue.getValue().getKey());
+
                 }
             } catch (HttpClientErrorException e) {
                 log.error("Error occurred while processing issue with key {}", jiraIssue.getKey(), e);
             }
         }
+    }
 
-        return ImmutableMap.of(
-                "new", newIssues,
-                "updated", updatedIssues,
-                "closed", closedIssues
-        );
+    private void createIssue(ScanRequest request, List<String> newIssues, Map.Entry<String, ScanResults.XIssue> xIssue, ScanResults.XIssue currentIssue) throws JiraClientException {
```

Comment:
method name should be more precise. Something like createIssueAndAddToNewIssuesList